code/game/objects/items/devices/laserpointer.dm

```diff
@@ -0,0 +1,218 @@
+				var/severity = (rand(0, 1) + diode.rating - eye_prot)
+				var/mob/living/carbon/human/H = C
+				var/obj/item/organ/internal/eyes/E = H.internal_organs_by_name[O_EYES]
+				if(!E)
+					outmsg = "<span class='notice'>You shine [src] at [C], but they don't seem to have eyes.</span>"
+					return
+
+				outmsg = "<span class='notice'>You shine [src] into [C]'s eyes.</span>"
+				switch(severity)
+					if(0)
+						//rank 1 diode with basic eye protection (like sunglasses), or rank 2 with industrial protection (like welding goggles)
+						to_chat(C, "<span class='info'>A small, bright dot appears in your vision.</span>")
+					if(1)
+						//rank 1 with no protection, rank 2 with basic protection, or rank 3 with industrial protection
+						to_chat(C, "<span class='notice'>Something bright flashes in the corner of your vision.</span>")
+					if(2)
+						//rank 1 or 2 with no protection, rank 2 or 3 with basic protection, or rank 3 with industrial protection
+						//alternatively, rank 1 with minor vulnerability (like night vision goggles)
+						flick("flash", C.flash_eyes())
+						to_chat(C, "<span class='danger'>A bright light shines across your eyes!</span>")
+					if(3)
+						//rank 1 with minor vulnerability, rank 2 or 3 with no protection, or rank 3 with basic protection
+						if(prob(3 * diode.rating))
+							C.Weaken(1)
+						flick("flash", C.flash_eyes())
+						E.damage += 1
+						to_chat(C, "<span class='danger'>A bright light briefly blinds you!</span>")
+					if(4)
+						//rank 3 with no protection, or rank 2 with minor vulnerability
+						if(prob(5 * diode.rating))
+							C.Weaken(1)
+						flick("e_flash", C.flash_eyes())
+						E.damage += 2
+						to_chat(C, "<span class='danger'>A blinding light burns your eyes!</span>")
+			else
+				outmsg = "<span class='notice'>You shine the [src] at [C], but miss their eyes.</span>"
+
+	//robots and AI
+	else if(issilicon(target))
+		var/mob/living/silicon/S = target
+		//20% chance to actually hit the sensors
+		if(prob(effectchance * diode.rating))
+			flick("flash", S.flash_eyes(affect_silicon = TRUE))
+			if (prob(3 * diode.rating))
+				S.Weaken(diode.rating)
+			to_chat(S, "<span class='warning'>Your sensors were blinded by a laser!</span>")
+			outmsg = "<span class='notice'>You blind [S] by shining [src] at their sensors.</span>"
+			add_attack_logs(user,S,"Tried disabling using [src]")
+		else
+			outmsg = "<span class='notice'>You shine the [src] at [S], but miss their sensors.</span>"
+
+	//cameras
+	else if(istype(target, /obj/machinery/camera))
+		var/obj/machinery/camera/C = target
+		if(prob(effectchance * diode.rating))
+			C.emp_act(4 - diode.rating)
+			outmsg = "<span class='notice'>You shine the [src] into the lens of [C].</span>"
+			add_attack_logs(user,C.name,"Tried disabling using [src]")
+		else
+			outmsg = "<span class='info'>You missed the lens of [C] with [src].</span>"
+
+	//cats!
+	for(var/mob/living/simple_animal/cat/C in view(1,targloc))
```

Comment:
`viewers()` only returns a list of mobs, slightly more appropriate here

Reply:
Change view to viewers, then?